cli/server/server.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/src-d/code-annotation/server/handler"
+	"github.com/src-d/code-annotation/server/repository"
+	"github.com/src-d/code-annotation/server/service"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"github.com/go-chi/cors"
+	"github.com/kelseyhightower/envconfig"
+)
+
+type appConfig struct {
+	Host     string `envconfig:"HOST"`
+	Port     int    `envconfig:"PORT" default:"8080"`
+	UIDomain string `envconfig:"UI_DOMAIN" default:"http://127.0.0.1:8080"`
+}
+
+func main() {
+	// main configuration
+	var conf appConfig
+	envconfig.MustProcess("", &conf)
+
+	// create repos
+	userRepo := &repository.Users{}
+
+	// create services
+	var oauthConfig service.OAuthConfig
+	envconfig.MustProcess("oauth", &oauthConfig)
+	oauth := service.NewOAuth(oauthConfig.ClientID, oauthConfig.ClientSecret)
+
+	var jwtConfig service.JWTConfig
+	envconfig.MustProcess("jwt", &jwtConfig)
+	jwt := service.NewJWT(jwtConfig.SigningKey)
+
+	// cors options
+	corsOptions := cors.Options{
+		AllowedOrigins: []string{"*"},
+		AllowedMethods: []string{"GET", "POST", "PUT", "OPTIONS"},
+		AllowedHeaders: []string{"Location", "Authorization", "Content-Type"},
+	}
+
+	// loger
+	logger := service.NewLogger()
+
+	r := chi.NewRouter()
+
+	ms := []func(http.Handler) http.Handler{
+		middleware.Recoverer,
+		cors.New(corsOptions).Handler,
+		middleware.RequestLogger(&middleware.DefaultLogFormatter{Logger: logger}),
+	}
```

Comment:
better write as in documentation:
```go
r := chi.NewRouter()
r.Use(middleware.Recoverer)
r.Use(cors.New(corsOptions).Handler)
r.Use(middleware.RequestLogger(&middleware.DefaultLogFormatter{Logger: logger}))

r.Get("/login", GetHandler(handler.Login(oauth)))
r.Get("/oauth-callback", GetHandler(handler.OAuthCallback(oauth, jwt, userRepo, conf.UIDomain, logger)))

r.Route("/api", func(r chi.Router) {
  r.Use(jwt.Middleware)
  ...
})
```

not because current code is wrong, but because it isn't that easy to understand.
Like I need to scan all code below this line for `ms`.

If you want to exclude static (though I don't see why, I think it should be logged also and `FrontendStatics` may panic also). You can wrap it in another `r.Group`. But `r.Use`, `r.Use` is much more readable than array.